airflow/exceptions.py

```diff
@@ -256,12 +278,21 @@ def __init__(
         self.method_name = method_name
         self.kwargs = kwargs
         self.timeout = timeout
+
         # Check timeout type at runtime
         if self.timeout is not None and not hasattr(self.timeout, "total_seconds"):
             raise ValueError("Timeout value must be a timedelta")
 
+        # Check keyword arguments
+        if kwargs:
+            if method_name is None:
+                raise ValueError("Keyword arguments not allowed when method is not specified")
+            else:
+                if "event" in kwargs:
+                    raise ValueError("Keyword argument must not contain reserved name 'event'")
```

Comment:
Shouldn't this  apply no mater what the method_name is?

Reply:
If `method_name` is unset then `kwargs` are disallowed period – so we don't bother checking them.